pkg/util/validation/limits.go

```diff
@@ -251,6 +252,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 
 	// Store-gateway.
 	f.IntVar(&l.StoreGatewayTenantShardSize, "store-gateway.tenant-shard-size", 0, "The tenant's shard size, used when store-gateway sharding is enabled. Value of 0 disables shuffle sharding for the tenant, that is all tenant blocks are sharded across all store-gateway replicas.")
+	f.BoolVar(&l.IgnoreNativeHistogramsOnRead, "store-gateway.ignore-native-histograms-on-read", false, "To stop processing native histograms as soon as possible, set to true")
```

Comment:
What are your plans on merging this to `main`? IMO if we want this PR to go to main soon, this default seems a little risky. Should we have them off by default?

Reply:
krajo also merged #4103, which just forwards histograms by default. So up to you folks how you choose to do this, I was just raising this as a point.